scopes/workspace/workspace/bit-map.ts

```diff
@@ -192,19 +192,24 @@ export class BitMap {
       const config = componentMap.config;
       if (!config) return;
       Object.keys(config).forEach((aspectId) => {
-        if (aspectId === sourceId.toString()) {
+        const fullSourceId = sourceId.toStringWithoutVersion();
+
+        const aspectIdWithoutVersion = ComponentID.fromString(aspectId).toStringWithoutVersion();
+        if (aspectIdWithoutVersion === fullSourceId) {
           config[targetId.toString()] = config[aspectId];
           delete config[aspectId];
           this.markAsChanged();
         }
+
         if (aspectId === EnvsAspect.id) {
           const envConfig = config[aspectId];
-          if (envConfig !== REMOVE_EXTENSION_SPECIAL_SIGN && envConfig.env === sourceId.toString()) {
+          if (envConfig !== REMOVE_EXTENSION_SPECIAL_SIGN && envConfig.env === fullSourceId) {
```

Comment:
The `envConfig.env` might have a value with a version, in that case, your new check will miss it.